src/land/afwa_snod2ioda.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+#
+# (C) Copyright 2020 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+import pygrib
+import time, os, sys
+import argparse
+import netCDF4 as nc
+import numpy as np
+from datetime import datetime, timedelta
+from pathlib import Path
+
+IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
+if not IODA_CONV_PATH.is_dir():
+    IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
+sys.path.append(str(IODA_CONV_PATH.resolve()))
+
+import ioda_conv_ncio as iconv
+from orddicts import DefaultOrderedDict
+
+vName = {
+    'A': "afwa_snow_depth",
+}
+
+locationKeyList = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("datetime", "string")
+]
+
+AttrData = {}
+
+
+class AFWA(object):
+
+    def __init__(self, filename, mask, writer):
+        self.filename = filename
+        self.mask = mask
+        self.data = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+        self.writer = writer
+        self._read()
+
+    def _read(self):
+        data = pygrib.open(self.filename)
+        lat, lon = data[1].latlons()
+        lons = lon[:].ravel()
+        lats = lat[:].ravel()
+        vals = data[1].values[:].ravel()
+        # defined errors and qc
+        errs = 0.15*vals
+
+        if self.mask == "maskout":
+            mask = np.logical_not(vals.mask)
+            vals = vals[mask]
+            lons = lons[mask]
+            lats = lats[mask]
+            errs = errs[mask]
+        # get global attributes
+        start_datetime = data[1].analDate
+        base_datetime = start_datetime.isoformat() + "Z"
+        # grbs.close()
+
+        valKey = vName['A'], self.writer.OvalName()
+        errKey = vName['A'], self.writer.OerrName()
+        qcKey = vName['A'], self.writer.OqcName()
+
+        for i in range(len(lons)):
+
+            locKey = lats[i], lons[i], base_datetime
+            self.data[0][locKey][valKey] = vals[i]
+            self.data[0][locKey][errKey] = errs[i]
+            self.data[0][locKey][qcKey] = 0
+
+            # write global attributes out
+            self.satellite = "afwa-merged"
```

Comment:
is this a 'satellite'? Do we need these attributes?

Reply:
No, it is a governmental agency (Air Force Weather Agency) rather than satellite. I am not sure if I need this information. Can I change satellite into agency, and then put AttrData["agency"] = AFWA?

Reply:
You can do whatever you want ;-) But to address @gthompsnJCSDA 's other comments, any metadata that you think is useful you should include as a general rule. Don't force yourself to use 'satellite' or 'sensor', but rather include what is relevant to this particular dataset. 'satellite' and 'sensor' are only really relevant for satellite-based radiances/retrievals/etc. observations.

Reply:
Got it. Thank you.  A good lesson learnt.